plugins/performance-lab/includes/site-health/audit-enqueued-assets/helper.php

```diff
@@ -190,120 +355,285 @@ function perflab_aea_enqueued_css_assets_test(): array {
+ * Gets the size of the asset in bytes.
  *
- * @since 1.0.0
+ * @since n.e.x.t
  *
- * @param string $resource_url URl resource link.
- * @return string Returns absolute path to the resource.
+ * @param string $resource_url URL of the resource.
+ * @return int|WP_Error Size of the asset in bytes or WP_Error if the request fails.
  */
-function perflab_aea_get_path_from_resource_url( string $resource_url ): string {
-	if ( '' === $resource_url ) {
-		return '';
+function perflab_aea_get_asset_size( string $resource_url ) {
+	$response = wp_remote_get(
+		$resource_url,
+		array(
+			'timeout' => 10,
+			'headers' => perflab_get_http_basic_authorization_headers(),
+		)
+	);
+
+	if ( is_wp_error( $response ) ) {
+		return $response;
+	}
+
+	if ( 200 !== wp_remote_retrieve_response_code( $response ) ) {
+		return new WP_Error(
+			'http_error',
+			wp_kses(
+				sprintf(
+					/* translators: %d is the HTTP status code, %s is the status header description */
+					__( 'Failed to retrieve the above asset with an HTTP status of <code>%1$d %2$s</code>.', 'performance-lab' ),
+					(int) wp_remote_retrieve_response_code( $response ),
+					esc_html( wp_remote_retrieve_response_message( $response ) )
+				),
+				array( 'code' => array() )
+			)
+		);
+	}
+
+	// TODO: A non-cacheable response should also be considered an error.
+	// TODO: A size of zero could be considered an error too.
+	return strlen( wp_remote_retrieve_body( $response ) );
+}
+
+/**
+ * Gets headers for HTTP Basic authorization headers.
+ *
+ * @since n.e.x.t
+ *
+ * @return array{ Authorization?: non-empty-string } Headers with copied Basic auth headers.
+ */
+function perflab_get_http_basic_authorization_headers(): array {
+	$headers = array();
+	if ( isset( $_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW'] ) ) {
+		$user                     = sanitize_text_field( wp_unslash( $_SERVER['PHP_AUTH_USER'] ) );
+		$pass                     = sanitize_text_field( wp_unslash( $_SERVER['PHP_AUTH_PW'] ) );
+		$headers['Authorization'] = 'Basic ' . base64_encode( $user . ':' . $pass ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode -- base64_encode() is used here to encode the credentials for forwarding basic auth headers.
 	}
+	return $headers;
+}
 
-	// Different content folder ex. /content/.
-	if ( 0 === strpos( $resource_url, content_url() ) ) {
-		return WP_CONTENT_DIR . substr( $resource_url, strlen( content_url() ) );
+/**
+ * Generates a table of blocking assets.
+ *
+ * @since n.e.x.t
+ *
+ * @param array{scripts: array<string, mixed>, styles: array<string, mixed>} $blocking_assets Array of blocking assets.
+ * @return string HTML table of blocking assets.
+ */
+function perflab_aea_generate_blocking_assets_table( array $blocking_assets ): string {
+	if ( 0 === count( $blocking_assets ) || ( isset( $blocking_assets['scripts'] ) && 0 === count( $blocking_assets['scripts'] ) && 0 === count( $blocking_assets['styles'] ) ) ) {
```

Comment:
Given that the `scripts` and `styles` keys are always set, this seems it can be simplified down to:

```suggestion
	if ( 0 === count( $blocking_assets['scripts'] ) && 0 === count( $blocking_assets['styles'] ) ) {
```